File: CyberJob.Core/Entities/Company.cs
namespace CyberJob.Core.Entities;

public class Company : BaseEntity
{
    public string? Image { get; set; }
    public string? BannerImage { get; set; }
    public string? Name { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public string? Address { get; set; }
    public string? ShortAddress { get; set; }
    public bool IsActive { get; set; }
    public bool IsVerified { get; set; }
    public DateOnly FoundingDate { get; set; }
    public string? About { get; set; }
    
    public int CategoryId { get; set; } 
    public virtual CompanyCategory? Category { get; set; }
    
}

File: internal/modules/company/domain/company.go
package domain

import (
	"time"

	"github.com/alberthaciverdiyev1/CyberJob/internal/platform/db"
)

type Company struct {
	db.BaseEntity
	Image        string `gorm:"type:varchar(255)"`
	BannerImage  string `gorm:"type:varchar(255)"`
	Name         string `gorm:"type:varchar(255);not null"`
	Email        string `gorm:"type:varchar(100);uniqueIndex"`
	Phone        string `gorm:"type:varchar(20)"`
	Address      string `gorm:"type:text"`
	ShortAddress string `gorm:"type:varchar(150)"`
	IsActive     bool   `gorm:"default:true"`
	IsVerified   bool   `gorm:"default:false"`
	FoundingDate *time.Time
	About        string `gorm:"type:text"`

	CategoryID uint `gorm:"not null"`
}
type CompanyFilter struct {
	CategoryID uint
	IsActive   *bool
	Name       string
	Limit      int
	Email      string
}
